lighthouse-core/test/computed/metrics/total-blocking-time-test.js

```diff
@@ -72,27 +63,28 @@ Object {
       const fcpTimeMs = 1500;
       const interactiveTimeMs = 2500;
 
-      expect(CumulativeLongQueuingDelay.calculateSumOfLongQueuingDelay(
-        events,
-        fcpTimeMs,
-        interactiveTimeMs
-      )).toBe(150);
+      expect(
+        TotalBlockingTime.calculateSumOfBlockingTime(events, fcpTimeMs, interactiveTimeMs)
+      ).toBe(150);
     });
 
-    it('clips queuing delay regions properly', () => {
-      const fcpTimeMs = 1050;
-      const interactiveTimeMs = 2050;
+    it('clips before finding blocking regions', () => {
+      const fcpTimeMs = 150;
+      const interactiveTimeMs = 300;
 
       const events = [
-        {start: 1000, end: 1110, duration: 110}, // Contributes 10ms.
-        {start: 2000, end: 2200, duration: 200}, // Contributes 50ms.
+        // The clipping is done first, so the task becomes [150, 200] after clipping and contributes
+        // 0ms of blocking time. This is in contrast to first calculating the blocking region ([100,
```

Comment:
would you mind adding a few other edge cases?

The fact that I can't figure out why this new version is different from before is my primary motivation :) 

I'm thinking...

```
          FCP at 1000 ms
          |
        =====  Task >50ms ends before 1050 ms
        ======  Task >50ms ends at 1050 ms
        =======  Task >50ms starts before 1000 ms and ends after 1050 ms

Reply:
Added these cases.